### DISCUSSION TITLE
Problem during installation ERROR: Dependency 'glib-2.0' tool variable 'glib_mkenums' contains erroneous value: '/usr/bin/glib-mkenums'

### DISCUSSION THREAD

**Discussion by ptrkpl** [end-4/dots-hyprland#615] [Q&A] [answered]
Hello,

During installation I managed to overcome several errors but I got stuck on this one.

`The Meson build system
Version: 1.4.1
Source dir: /home/ptrk/Downloads/dots-hyprland/cache/ags
Build dir: /home/ptrk/Downloads/dots-hyprland/cache/ags/build
Build type: native build
WARNING: You should add the boolean check kwarg to the run_command call.
         It currently defaults to false,
         but it will default to true in future releases of meson.
         See also: https://github.com/mesonbuild/meson/issues/9300
Project name: ags
Project version: 1.8.2
C compiler for the host machine: cc (gcc 14.1.1 "cc (GCC) 14.1.1 20240522")
C linker for the host machine: cc ld.bfd 2.42.0
Host machine cpu family: x86_64
Host machine cpu: x86_64
Found pkg-config: YES (/usr/bin/pkg-config) 2.1.1
Run-time dependency glib-2.0 found: YES 2.80.3
Run-time dependency gobject-introspection-1.0 found: YES 1.80.1
Run-time dependency gio-2.0 found: YES 2.80.3
Run-time dependency gjs-1.0 found: YES 1.80.2
Run-time dependency gtk+-3.0 found: YES 3.24.42

Executing subproject gvc

gvc| Project name: gvc
gvc| Project version: undefined
gvc| C compiler for the host machine: cc (gcc 14.1.1 "cc (GCC) 14.1.1 20240522")
gvc| C linker for the host machine: cc ld.bfd 2.42.0
gvc| Dependency glib-2.0 found: YES 2.80.3 (cached)
gvc| Program /usr/bin/glib-mkenums found: NO

subprojects/gvc/meson.build:30:21: ERROR: Dependency 'glib-2.0' tool variable 'glib_mkenums' contains erroneous value: '/usr/bin/glib-mkenums'

This is a distributor issue -- please report it to your glib-2.0 provider.

A full log can be found at /home/ptrk/Downloads/dots-hyprland/cache/ags/build/meson-logs/meson-log.txt
[./update-ags.sh]: Command "meson setup build --reconfigure" has failed.
You may need to resolve the problem manually BEFORE repeating this command.`

I do not know how to debug this problem

**Comment by clsty** [comment]
`rm -r /home/ptrk/Downloads/dots-hyprland/cache/ags` and try again.

**Reply by ptrkpl** [reply]
> `rm -r /home/ptrk/Downloads/dots-hyprland/cache/ags` and try again.

Done, and it didn't help

**Reply by clsty** [reply]
So, the problem here is that AGS can't be built on your computer.

According to https://aylur.github.io/ags-docs/config/installation/ , you may try install dependencies first: 
```bash
sudo pacman -S typescript npm meson gjs gtk3 gtk-layer-shell gnome-bluetooth-3.0 upower networkmanager gobject-introspection libdbusmenu-gtk3 libsoup3
```

PS. Though, be sure NOT to install AGS without version control (e.g. from AUR). We're doing version control for this repo to protect it against possible breaking changes from AGS.

**Reply by ptrkpl** [reply]
I have all these packages already:
 sudo pacman -S typescript npm meson gjs gtk3 gtk-layer-shell gnome-bluetooth-3.0 upower networkmanager gobject-introspection libdbusmenu-gtk3 libsoup3
[sudo] password for ptrk:
warning: typescript-5.4.5-1 is up to date -- reinstalling
warning: npm-10.8.1-1 is up to date -- reinstalling
warning: meson-1.4.1-1 is up to date -- reinstalling
warning: gjs-2:1.80.2-2.1 is up to date -- reinstalling
warning: gtk3-1:3.24.42-1.1 is up to date -- reinstalling
warning: gtk-layer-shell-0.8.2-1 is up to date -- reinstalling
warning: gnome-bluetooth-3.0-46.0-1 is up to date -- reinstalling
warning: upower-1.90.4-1.1 is up to date -- reinstalling
warning: networkmanager-1.48.0-1 is up to date -- reinstalling
warning: gobject-introspection-1.80.1-3.2 is up to date -- reinstalling
warning: libdbusmenu-gtk3-16.04.0.r498-2 is up to date -- reinstalling
warning: libsoup3-3.4.4-1.1 is up to date -- reinstalling
resolving dependencies...
looking for conflicting packages...

**Comment by phoenixpev** [comment, marked as answer]
I've just had this exact problem - I ran `yay -S aylurs-gtk-shell` separately and tried again

**Reply by ptrkpl** [reply]
that actually helped - Thanks a lot!

**Reply by clsty** [reply]
Welp, this is just the AUR package that lack version controlling required by this repo.
Which means that it may not work properly.

Though since this AUR package works for you, I guess we need to consider copying the idea of its PKGBUILD.

**Reply by end-4** [reply]
> Which means that it may not work properly.

eh should be fine ags is pretty stable now
aylur is cooking something else

**Comment by clsty** [comment]
The key reason here is that `glib2` has been separated into 2 packages recently: `glib2` and `glib2-devel`. So you also need to install `glib2-devel` as dependency.

We're now using a local PKGBUILD for AGS installation which includes `glib2-devel` as dependency so this problem is solved.
